### PR TITLE
Add capability to get pod labels

### PR DESCRIPTION
**Pull request by tharinduwijewardane**
Fixes: https://github.com/pixie-io/pixie/issues/449

Example script:
```
import px
df = px.DataFrame(table='http_events', start_time='-10s')
df.labels = px.pod_id_to_pod_labels(df.ctx['pod_id'])
px.display(df)
```

Output:
Screenshot from 2022-06-22 23-48-42

### CODE REVIEW

**Comment by zasgar** [comment]
ok to test

**Review by zasgar — CHANGES_REQUESTED** [review]

**Comment by zasgar** [comment]
Thanks a lot for the contribution. This looks great. There is a just a small bug with one of the tests and then it should be good to go.

File: src/carnot/funcs/metadata/metadata_ops_test.cc [hunk 1]
```diff
@@ -105,6 +105,14 @@ TEST_F(MetadataOpsTest, pod_id_to_pod_name_test) {
   udf_tester.ForInput("missing").Expect("");
 }
 
+TEST_F(MetadataOpsTest, pod_id_to_pod_labels_test) {
+  auto function_ctx = std::make_unique<FunctionContext>(metadata_state_, nullptr);
+  auto udf_tester = px::carnot::udf::UDFTester<PodIDToPodLabelsUDF>(std::move(function_ctx));
+  udf_tester.ForInput("1_uid").Expect("{\"k1\":\"v1\", \"k2\":\"v2\"}");
+  udf_tester.ForInput("2_uid").Expect("{\"k1\":\"v1\"");
```
[the remark this answers is not in the document]
Comment (tharinduwijewardane) [reply src/carnot/funcs/metadata/metadata_ops_test.cc]: Fixed

**Comment by zasgar** [comment]
Ok to test

File: src/vizier/services/metadata/controllers/k8smeta/k8s_metadata_handler.go [hunk 2]
```diff
@@ -938,13 +939,16 @@ func getResourceUpdateFromPod(pod *metadatapb.Pod, uv int64) *metadatapb.Resourc
 		hostname = pod.Spec.Hostname
 	}
 
+	podLabels, _ := json.Marshal(pod.Metadata.Labels)
+
```
Comment (vihangm) [inline src/vizier/services/metadata/controllers/k8smeta/k8s_metadata_handler.go:943]: you need a null check here. if `pod.Metadata.Labels` is null, then `podLabels` will be the literal string `"null"`

File: src/vizier/services/metadata/controllers/k8smeta/k8s_metadata_handler_test.go [hunk 3]
```diff
@@ -981,6 +981,7 @@ func TestPodUpdateProcessor_GetUpdatesToSend(t *testing.T) {
 					UID:              "ijkl",
 					Name:             "object_md",
 					Namespace:        "",
+					Labels:           `{"app":"myApp1", "project":"myProj1"}`,
```
Comment (vihangm) [inline src/vizier/services/metadata/controllers/k8smeta/k8s_metadata_handler_test.go:984]: the test fixture for this test doesn't have labels, so this test fails. can you please fix this test?
you can run `bazel test //src/vizier/services/metadata/controllers/k8smeta:k8smeta_test` to run the test locally
`bazel test //src/vizier/services/metadata/controllers/k8smeta:k8smeta_test --test_output=streamed` is more convenient when actively developing/fixing.

**Review by vihangm — CHANGES_REQUESTED** [review]
This looks great. Need a couple of small changes to fix a broken test and then we can merge this in!

File: src/vizier/services/metadata/controllers/k8smeta/k8s_metadata_handler.go (hunk 2 above)
Comment (tharinduwijewardane) [reply src/vizier/services/metadata/controllers/k8smeta/k8s_metadata_handler.go]: Fixed with https://github.com/pixie-io/pixie/pull/468/commits/233860db09eecc9fb6827bae2eef3b6eb8f48926

File: src/vizier/services/metadata/controllers/k8smeta/k8s_metadata_handler_test.go (hunk 3 above)
Comment (tharinduwijewardane) [reply src/vizier/services/metadata/controllers/k8smeta/k8s_metadata_handler_test.go]: Fixed with https://github.com/pixie-io/pixie/pull/468/commits/4c274415c0861836c56b3a4bf08a5da08e612ea0
Thanks for the guidance

**Review by vihangm — APPROVED** [review]

**Pull request closed without merging**